service/controller/clusterapi/v29/adapter/guest_subnets.go

```diff
@@ -44,30 +44,30 @@ func (s *GuestSubnetsAdapter) Adapt(cfg Config) error {
 		}
 	}
 
-	for i, az := range zones {
-		snetName := key.PublicSubnetName(i)
+	for _, az := range zones {
+		snetName := key.SanitizeCFResourceName(key.PublicSubnetName(az.Name))
 		snet := Subnet{
 			AvailabilityZone:    az.Name,
 			CIDR:                az.Subnet.Public.CIDR,
 			Name:                snetName,
 			MapPublicIPOnLaunch: false,
 			RouteTableAssociation: RouteTableAssociation{
-				Name:           key.PublicSubnetRouteTableAssociationName(i),
-				RouteTableName: "PublicRouteTable",
+				Name:           key.SanitizeCFResourceName(key.PublicSubnetRouteTableAssociationName(az.Name)),
+				RouteTableName: key.SanitizeCFResourceName(key.PublicRouteTableName(key.MasterAvailabilityZone(cfg.CustomObject))),
```

Comment:
:question: Should this be hooked to master's AZ or should this be for corresponding AZ?

Reply:
We still need to clear this up but this is a follow up step IMO.